### PR TITLE
Support saml-group-membership-mapper

### PR DESCRIPTION
**Pull request by mattock**

### CODE REVIEW

**Comment by mattock** [comment]
@treydock let me know if you want some additional tests for this. I was not sure if those would be needed.

**Comment by treydock** [comment]
Looks like failures are style related from rubocop.

File: lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb [hunk 1]
```diff
@@ -46,10 +46,10 @@ def self.instances
             protocol_mapper[:claim_name] = d['config']['claim.name']
             protocol_mapper[:json_type_label] = d['config']['jsonType.label']
           end
-          if protocol_mapper[:type] == 'oidc-group-membership-mapper'
+          if ['oidc-group-membership-mapper','saml-group-membership-mapper'].include?(protocol_mapper[:type])
```
Comment (treydock) [inline lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb:49]:
```suggestion
          if ['oidc-group-membership-mapper', 'saml-group-membership-mapper'].include?(protocol_mapper[:type])
```

File: lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb [hunk 2]
```diff
@@ -109,10 +109,10 @@ def create
       data[:config][:'claim.name'] = resource[:claim_name] if resource[:claim_name]
       data[:config][:'jsonType.label'] = resource[:json_type_label] if resource[:json_type_label]
     end
-    if resource[:type] == 'oidc-group-membership-mapper'
+    if ['oidc-group-membership-mapper','saml-group-membership-mapper'].include?(resource[:type])
```
Comment (treydock) [inline lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb:112]:
```suggestion
    if ['oidc-group-membership-mapper', 'saml-group-membership-mapper'].include?(resource[:type])
```

File: lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb [hunk 3]
```diff
@@ -193,10 +193,10 @@ def flush
         config[:'claim.name'] = resource[:claim_name] if resource[:claim_name]
         config[:'jsonType.label'] = resource[:json_type_label] if resource[:json_type_label]
       end
-      if resource[:type] == 'oidc-group-membership-mapper'
+      if ['oidc-group-membership-mapper','saml-group-membership-mapper'].include?(resource[:type])
```
Comment (treydock) [inline lib/puppet/provider/keycloak_protocol_mapper/kcadm.rb:196]:
```suggestion
      if ['oidc-group-membership-mapper', 'saml-group-membership-mapper'].include?(resource[:type])
```

**Comment by treydock** [comment]
Released with v6.23.0

**Pull request merged**